lib/flux-core/src/iter/range.rs

```diff
@@ -1,15 +1,8 @@
-#![allow(unused)]
-#![feature(step_trait)]
-
-use flux_rs::attrs::*;
-
-extern crate flux_core;
-
-use std::iter::Step;
+use flux_attrs::*;
 
 // TODO(RJ): use default spec `true` for `can_step_forward` and `can_step_backward`
 
-#[extern_spec(core::ops)]
+#[extern_spec(core::iter)]
```

Comment:
I think we should document these associated refinements. It's not clear to me what each one is supposed to mean. Like, why can a `usize` always step forward? Also, I'm a bit hesitant to have associated refinements that don't have default bodies because that will break any implementation that we have not yet refined.

Maybe we should also use these associated refinements to refine the methods in the trait to sanity check them.

Reply:
The `Step` trait has a couple of methods, and the associated refinements are not used to specify them. I understand that the end goal is to specify iterators, but the implementation of `Iterator` for `Range` ultimately uses the methods in `Step`, so one should be able to specify them with the associated refinements.

The comment about documentation is more about me not being able to say if these associated refinements make sense. But perhaps this is more about this PR not standing on its own, because I need to see how it interacts with `Iterator`.

The UIF is an interesting idea; it also crossed my mind. But depending on how one specify things it may also break, but in the sense that you won't be able to use the methods, because you don't have enough information.

Reply:
Was just looking at the _actual_ `Step` in core, btw, they have a lot of specs explicitly spelled out.

Reply:
Ok I think I dealt with (1), (2) and (3), using UIFs, adding comment and tweaking the specs.

Ultimately, would be cool to actually validate against the real implementations in `core`!